lib/nerves_runtime/kv.ex

```diff
@@ -123,4 +156,24 @@ defmodule Nerves.Runtime.KV do
     |> Enum.map(fn {k, v} -> {String.replace_leading(k, active, ""), v} end)
     |> Enum.into(%{})
   end
+
+  defp read_config(file) do
+    case File.read(file) do
+      {:ok, config} -> {:ok, config}
+      _ -> {:error, :no_config}
+    end
+  end
+
+  defp parse_config(config) do
+    [config] =
+      config
+      |> String.split("\n", trim: true)
+      |> Enum.map(&String.trim/1)
+      |> Enum.reject(&String.starts_with?(&1, "#"))
+
+    [dev_name, <<"0x", dev_offset::binary()>>, <<"0x", env_size::binary()>> | _] =
```

Comment:
Ugh (not on you, ugh on Elixir). It's not guaranteed that the number be hex in this file. I think that you need:

```elixir
defp parse_int(<<"0x", hex_int>>), do: String.to_integer(hex_int, 16)
defp parse_int(decimal_int), do: String.to_integer(decimal_int)
```